static/app/components/compactSelect/control.tsx

```diff
@@ -313,6 +320,18 @@ export function Control({
     },
   });
 
+  /**
+   * The menu's full width, before any option has been filtered out. Used to maintain a
+   * constant width while the user types into the search box.
+   */
+  const [menuFullWidth, setMenuFullWidth] = useState<number>();
+  useLayoutEffect(() => {
+    if (!overlayRef.current || !overlayIsOpen) {
+      return;
+    }
+    setMenuFullWidth(overlayRef.current?.offsetWidth);
+  }, [overlayRef, overlayIsOpen]);
```

Comment:
`overlayRef` is, as the name suggests, a ref object and doesn't change reference throughout `Control`'s lifecycle, but eslint doesn't recognize that and will complain if I remove it from the deps array. Including it has no impact on how often this effect is called.